libp2p/connmanager.nim

```diff
@@ -67,7 +67,8 @@ type
     muxed: Table[PeerId, seq[Muxer]]
     connEvents: array[ConnEventKind, OrderedSet[ConnEventHandler]]
     peerEvents: array[PeerEventKind, OrderedSet[PeerEventHandler]]
-    expectedConnectionsOverLimit*: Table[(PeerId, Direction), Future[Muxer]]
+    expectedConnectionsOverLimit*:
+      Table[(PeerId, Direction), Future[Muxer].Raising([CancelledError])]
```

Comment:
What would be the difference between this and using a `Future[Result[Muxer, ...]]`.

Actually, I guess the broader question would be: Is there a reason we're going for exceptions over handling errors with `Result`?

Reply:
I'm not an expert, but Nim natively supports exceptions and there's also compiler support for pros signatures https://nim-lang.org/docs/tut2.html#exceptions-annotating-procs-with-raised-exceptions. `Result`was created in some status library. It might be painful to use `Result`if Nim std wasn't designed for that. I can't say much as in general, this project uses more exceptions than `Result`.